### ISSUE TITLE
How can I do evaluation by group in xgb.cv?

### ISSUE THREAD

**Issue by FingerDrill** [dmlc/xgboost#2600]
If I want to calculate mean f1 score with user_id, how to calculate it?

Customized evaluation **function like below works well in xgb.train** :
```
xgb_eval_f1 <- function (yhat, dtrain) {
   require(ModelMetrics)
   y = getinfo(dtrain, "label")
   dt <- data.table(user_id=train[user_id %in% val_users, user_id], purch=y, pred=yhat)
   f1 <- mean(dt[,.(f1score=f1Score(purch, pred, cutoff=0.2)), by=user_id]$f1score)
   return (list(metric = "f1", value = f1))
}
```
in this code, validation set is selected by user with 'val_users'.

**If I use the function in xgb.cv, it makes r session terminated.**

In the function, user_id can be added to each row. But in xgb.cv, I can't add the user_id information because it makes folds.

I've searched about info in xgb.DMatrix and customizing of evaluation function but I couldn't find solution.
I think adding new attribute to dtrain may not work because attribute are not attached to each instance so when I use xgb.cv, it have no sense.**(Is added attribute sliced when xgb.cv makes folds?)**

## **Question summary :**
**Is there any solution to attach each instance with new information(ID, etc.) like 'label' or 'weight' in info but not used to train?**

**Comment by FingerDrill** [comment]
I find solution at #86 (.. How can I reference #86? it doesn't appear.)
But I'm still fighting against session termination. :(
